ext/ftp/ftp.c

```diff
@@ -1931,6 +1931,11 @@ static void ftp_ssl_shutdown(ftpbuf_t *ftp, php_socket_t fd, SSL *ssl_handle) {
 					/* SSL wants a write. Really odd. Let's bail out. */
 					done = 1;
 					break;
+				case SSL_ERROR_SYSCALL:
+					/* Something went totally wrong; bail out to avoid raising
+					   a spurious warning. */
```

Comment:
Might be worth to mention that it's likely due the peer (server) closing the connection without sending close_notify shutdown alert. Think it might be often the reason for this.

Reply:
Yes, this is a good idea.